tests/pybind11_tests.h

```diff
@@ -69,3 +70,15 @@ template<> class type_caster<RValueCaster> {
 };
 PYBIND11_NAMESPACE_END(detail)
 PYBIND11_NAMESPACE_END(pybind11)
+
+template <typename F>
+void ignoreOldStyleInitWarnings(F &&body) {
+    py::exec(R"(
```

Comment:
Comment from #2746:

- @YannickJadoul: For reference, this was the old code, without `py::exec` (and thus without requiring `<pybind11/eval.h>` to be included in `pybind11_tests.h` and thus in all tests).

  @bstaletic slightly prefers this [new version], and I think I do to. However, curious to learn what other reviewers think.

  ```c++
  /// Simplified ``with warnigns.catch_warnings()`` wrapper
  template <typename F>
  void ignoreOldStyleInitWarnings(F &&body) {
      auto message = "pybind11-bound class '.+' is using an old-style placement-new '(?:__init__|__setstate__)' which has been deprecated";
      auto category = py::reinterpret_borrow<py::object>(PyExc_FutureWarning);
      auto warnings = py::module_::import("warnings");
      auto context_mgr = warnings.attr("catch_warnings")();
      context_mgr.attr("__enter__")();
      warnings.attr("filterwarnings")("ignore", py::arg("message")=message, py::arg("category")=category);
      body();
      // Exceptions in `body` not handled; see PEP 343 when these would need to be added
      context_mgr.attr("__exit__")(py::none(), py::none(), py::none());
  }
  ```